packages/starknet-snap/src/types/snapState.ts

```diff
@@ -10,19 +10,28 @@ export type SnapState = {
   transactionRequests?: TransactionRequest[];
 };
 
+export type FormattedCallData = {
+  contractAddress: string;
+  calldata?: string[];
+  entrypoint: string;
+  isTransfer: boolean; // Flag to indicate if this call is a transfer
+  transferSenderAddress?: string;
+  transferRecipientAddress?: string;
+  transferAmount?: string;
+  transferTokenDecimals?: number;
+  transferTokenSymbol?: string;
+};
```

Comment:
nit: 

if you wanna to mute the type issue and reduce the checking for transfer , you can do it like this

and you can just check if tokenTransferData is undefined or not 

and you can even reuse the `TokenTransferData` in somewhere

```suggestion
export type TokenTransferData = {
     senderAddress: string,
     recipientAddress: string,
     amount:string,
     decimal:number,
     symbol:string     
}

export type FormattedCallData = {
  contractAddress: string;
  calldata?: string[];
  entrypoint: string;
  tokenTransferData?: TokenTransferData
};
```